src/otx/algo/detection/utils/yolov7_v9_utils.py

```diff
@@ -0,0 +1,256 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Utils for YOLOv7 and v9.
+
+Reference : https://github.com/WongKinYiu/YOLO
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import torch
+from einops import rearrange
+from torch import Tensor, nn
+from torchvision.ops import batched_nms
+
+if TYPE_CHECKING:
+    from otx.algo.detection.detectors import SingleStageDetector
+
+
+def auto_pad(kernel_size: int | tuple[int, int], dilation: int | tuple[int, int] = 1, **kwargs) -> tuple[int, int]:  # noqa: ARG001
+    """Auto Padding for the convolution blocks.
+
+    Args:
+        kernel_size (int | tuple[int, int]): The kernel size of the convolution block.
+        dilation (int | tuple[int, int]): The dilation rate of the convolution block. Defaults to 1.
+
+    Returns:
+        tuple[int, int]: The padding size for the convolution block.
+    """
+    if isinstance(kernel_size, int):
+        kernel_size = (kernel_size, kernel_size)
+    if isinstance(dilation, int):
+        dilation = (dilation, dilation)
+
+    pad_h = ((kernel_size[0] - 1) * dilation[0]) // 2
+    pad_w = ((kernel_size[1] - 1) * dilation[1]) // 2
+    return pad_h, pad_w
+
+
+def round_up(x: int | Tensor, div: int = 1) -> int | Tensor:
+    """Rounds up `x` to the bigger-nearest multiple of `div`.
+
+    Args:
+        x (int | Tensor): The input value.
+        div (int): The divisor value. Defaults to 1.
+
+    Returns:
+        int | Tensor: The rounded up value.
+    """
+    return x + (-x % div)
+
+
+def generate_anchors(image_size: tuple[int, int], strides: list[int]) -> tuple[Tensor, Tensor]:
```

Comment:
We have AnchorGenerator, isn't it possible to use its functionality to generate anchors? Or maybe we could change AnchorGenerator to align with YOLO interface

Reply:
Thanks for your suggestion, but I think it seems to take a time for this.
Can I keep this now and check it later? I left TODO comment at 960f3d5.